File: refactor-supabase.py
import os
import re

src_dir = os.path.join(os.path.dirname(__file__), 'systems', 'production')
sp = r'(?:productionSupabase|productionDispatchSupabase|productionPurchaseSupabase|productionInventorySupabase|prodSupabase)'

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    original = content

    # Replace delete
    content = re.sub(
        r'await\s+' + sp + r'\s*\.\s*from\(\s*[\'"]([^\'"]+)[\'"]\s*\)\s*\.\s*delete\(\)\s*\.\s*eq\(\s*[\'"]id[\'"]\s*,\s*([^)]+)\)',
        r"await productionApi.delete('\1', \2)",
        content
    )
    
    # Replace patch
    content = re.sub(
        r'await\s+' + sp + r'\s*\.\s*from\(\s*[\'"]([^\'"]+)[\'"]\s*\)\s*\.\s*update\(([^)]+)\)\s*\.\s*eq\(\s*[\'"]id[\'"]\s*,\s*([^)]+)\)',
        r"await productionApi.patch('\1', \3, \2)",
        content
    )

    # Replace insert
    content = re.sub(
        r'await\s+' + sp + r'\s*\.\s*from\(\s*[\'"]([^\'"]+)[\'"]\s*\)\s*\.\s*insert\(([^)]+)\)(?:\s*\.\s*select\(\))?(?:\s*\.\s*single\(\))?',
        r"await productionApi.post('\1', \2)",
        content
    )

    # Replace get (without await)
    content = re.sub(
        sp + r'\s*\.\s*from\(\s*[\'"]([^\'"]+)[\'"]\s*\)\s*\.\s*select\(\s*[\'"][^\'"]*[\'"]\s*\)(?:\s*\.\s*order\([^)]+\))?',
        r"productionApi.get('\1')",
        content
    )

    # Clean imports
    content = re.sub(r'import \{.*?production.*?Supabase.*?\} from [\'"]@/systems/production/supabase[\'"];', 'import { productionApi } from "@/systems/production/lib/api";\nimport { API_URL, getToken } from "@/lib/auth";', content)

    if content != original:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print("Updated", filepath)

for root, dirs, files in os.walk(src_dir):
    for f in files:
        if f.endswith(('.tsx', '.ts', '.js')):
            process_file(os.path.join(root, f))

print("Done")

File: refactor-supabase.js
const fs = require('fs');
const path = require('path');

const srcDir = path.join(__dirname, 'systems', 'production');

function processFile(filePath) {
  let content = fs.readFileSync(filePath, 'utf8');
  let originalContent = content;

  // Add imports if needed
  if (content.includes('productionSupabase') && !content.includes('API_URL')) {
    content = content.replace(/import \{ productionSupabase[^}]*\} from [^;]+;/, 'import { API_URL, getToken } from "@/lib/auth";\n$&');
  }

  // Handle .select()
  // e.g. await productionSupabase.from('table').select('*')
  content = content.replace(
    /(?:await\s+)?productionSupabase\.from\(['"]([^'"]+)['"]\)\.select\([^)]*\)/g,
    `await fetch(\`\${API_URL}/production/$1\`, { headers: { Authorization: \`Bearer \${getToken()}\` } }).then(res => res.json()).then(res => ({ data: res.data, error: res.success === false ? res.message : null }))`
  );
  content = content.replace(
    /(?:await\s+)?productionDispatchSupabase\.from\(['"]([^'"]+)['"]\)\.select\([^)]*\)/g,
    `await fetch(\`\${API_URL}/production/$1\`, { headers: { Authorization: \`Bearer \${getToken()}\` } }).then(res => res.json()).then(res => ({ data: res.data, error: res.success === false ? res.message : null }))`
  );

  // Handle .insert(data)
  // e.g. await productionSupabase.from('table').insert([data])
  content = content.replace(
    /(?:await\s+)?productionSupabase\.from\(['"]([^'"]+)['"]\)\.insert\(([^)]+)\)/g,
    `await fetch(\`\${API_URL}/production/$1\`, { method: 'POST', headers: { 'Content-Type': 'application/json', Authorization: \`Bearer \${getToken()}\` }, body: JSON.stringify($2) }).then(res => res.json()).then(res => ({ data: res.data, error: res.success === false ? res.message : null }))`
  );

  // Handle .update(data).eq('id', id)
  // e.g. await productionSupabase.from('table').update({ status }).eq('id', id)
  // This is hard to regex perfectly. Let's do a basic one.
  content = content.replace(
    /(?:await\s+)?productionSupabase\.from\(['"]([^'"]+)['"]\)\.update\(([^)]+)\)\.eq\(['"]id['"],\s*([^)]+)\)/g,
    `await fetch(\`\${API_URL}/production/$1/\${$3}\`, { method: 'PATCH', headers: { 'Content-Type': 'application/json', Authorization: \`Bearer \${getToken()}\` }, body: JSON.stringify($2) }).then(res => res.json()).then(res => ({ data: res.data, error: res.success === false ? res.message : null }))`
  );

  // Handle .delete().eq('id', id)
  content = content.replace(
    /(?:await\s+)?productionSupabase\.from\(['"]([^'"]+)['"]\)\.delete\(\)\.eq\(['"]id['"],\s*([^)]+)\)/g,
    `await fetch(\`\${API_URL}/production/$1/\${$2}\`, { method: 'DELETE', headers: { Authorization: \`Bearer \${getToken()}\` } }).then(res => res.json()).then(res => ({ data: res.data, error: res.success === false ? res.message : null }))`
  );

  // Remove the supabase imports if they are no longer used
  content = content.replace(/import \{ productionSupabase[^}]*\} from [^;]+;/, '');

  if (content !== originalContent) {
    fs.writeFileSync(filePath, content, 'utf8');
    console.log(`Updated ${filePath}`);
  }
}

function walkDir(dir) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const fullPath = path.join(dir, file);
    if (fs.statSync(fullPath).isDirectory()) {
      walkDir(fullPath);
    } else if (fullPath.endsWith('.tsx') || fullPath.endsWith('.js') || fullPath.endsWith('.ts')) {
      processFile(fullPath);
    }
  }
}

walkDir(srcDir);
console.log('Done refactoring Supabase calls.');
